### PR TITLE
Make private fields protected in QueryManager

### PR DESCRIPTION
**Pull request by alessbell**
This PR makes all private fields/methods on the `QueryManager` class protected, making them accessible to any class that extends `QueryManager` while remaining inaccessible externally.

The main benefit for our purposes is being able to access type information in the [`NextSSRApolloClient` class](https://github.com/apollographql/apollo-client-nextjs/blob/main/package/src/ssr/NextSSRApolloClient.tsx) which (unsafely) accesses `QueryManager` methods.

Without going so far as to make these fields public, `protected` seems like a reasonable middle ground so we can still benefit from TS types in this scenario.

## Before
CleanShot 2023-06-26 at 11 53 51

## After
Note the `number` annotation in the D.TS output for `foo`:
CleanShot 2023-06-26 at 11 53 58

### Checklist:

- [x] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [x] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Comment by jerelmiller** [comment]
Could we only move over the members that are accessed by `NextSSRApolloClient`? There is still a lot of `QueryManager` that will only ever need to be private, so I'd love to keep those that way as much as possible.

**Comment by alessbell** [comment]
@jerelmiller sure, that works too - updated 👍

File: src/core/QueryManager.ts [hunk 1]
```diff
@@ -144,7 +146,7 @@ export class QueryManager<TStore> {
     assumeImmutableResults?: boolean;
   }) {
     const defaultDocumentTransform = new DocumentTransform(
-      (document) => this.cache.transformDocument(document),
```
Comment (jerelmiller) [inline src/core/QueryManager.ts:147]: I really need to fix my editor so that it removes trailing whitespace 🤦‍♂️

**Review by jerelmiller — APPROVED** [review]
Appreciate you scoping this down! LGTM

File: src/core/QueryManager.ts (hunk 1 above)
Comment (phryneas) [reply src/core/QueryManager.ts:147]: Prettier will take care of that soon :)

**Comment by phryneas** [comment]
I'm gonna merge this since Alessia is OOO, and there's no good reason to let this sit around any longer :)

**Pull request merged**